### PR TITLE
Change saveat handling similar to DiffEq

### LINKED ISSUE #17: saveat behavior different from DiffEq algorithms
MWE:

```
using MATLABDiffEq, OrdinaryDiffeq

function test!(du, u, p, t)
    du[1] = -10 * u[1]
    du[2] = -0.01 * u[2]
end
u0 = [1.01;1.01]
tspan = (0.0, 1.0)
prob = ODEProblem(test!, u0, tspan)
 
julia> sol = solve(prob,MATLABDiffEq.ode15s(),saveat=0.1)
retcode: Default
Interpolation: 1st order linear
t: 3-element Array{Float64,1}:
 0.0
 0.1
 1.0
u: 3-element Array{Array{Float64,1},1}:
 [1.01, 1.01]
 [0.37077117078064675, 1.0089905071455934]
 [4.5854039847379484e-5, 0.999950334377818]

julia> sol = solve(prob,QNDF(),saveat=0.1)
retcode: Success
Interpolation: 1st order linear
t: 11-element Array{Float64,1}:
 0.0
 0.1
 0.2
 0.3
 0.4
 0.5
 0.6
 0.7
 0.8
 0.9
 1.0
u: 11-element Array{Array{Float64,1},1}:
 [1.01, 1.01]
 [0.3704455272364188, 1.0089905040249973]
 [0.13627476141582867, 1.0079820112760844]
 [0.050307801181049064, 1.0069745370891778]
 [0.01851968870439673, 1.0059680647903897]
 [0.006835631288839779, 1.0049625956099648]
 [0.0025256513813332842, 1.003958139676276]
 [0.0009254926520955427, 1.0029546814523527]
 [0.00034266531842009777, 1.001952223812029]
 [0.00012726307534597449, 1.0009507730292209]
 [4.611900409656228e-5, 0.9999503229774236]
```

### PR DESCRIPTION
**Pull request by utkarsh530**
Fixes #17

### CODE REVIEW

File: src/MATLABDiffEq.jl [hunk 1]
```diff
@@ -29,7 +29,12 @@ function DiffEqBase.__solve(
     f = prob.f
     u0 = prob.u0
 
-    tspan = sort(unique([prob.tspan[1];saveat;prob.tspan[2]]))
+    if typeof(saveat) <: Number
+        tspan = [prob.tspan[1]:saveat:prob.tspan[2];]
```
Comment (ChrisRackauckas) [inline src/MATLABDiffEq.jl:33]: `tspan = Array(prob.tspan[1]:saveat:prob.tspan[2])`

Comment (ChrisRackauckas) [inline src/MATLABDiffEq.jl:33]:
```suggestion
        tspan = Array(prob.tspan[1]:saveat:prob.tspan[2])
```

**Comment by ChrisRackauckas** [comment]
Any other planned changes, or should we tag? If we should tag, increment it by a patch.

**Comment by utkarsh530** [comment]
Is it good have some stats like this?:

```
julia> sol = solve(prob,MATLABDiffEq.ode15s(),saveat=0.3)
42 successful steps
1 failed attempts
57 function evaluations
1 partial derivatives
10 LU decompositions
53 solutions of linear systems
retcode: Default
Interpolation: 1st order linear
t: 5-element Array{Float64,1}:
 0.0
 0.3
 0.6
 0.9
 1.0
u: 5-element Array{Array{Float64,1},1}:
 [1.01, 1.01]
 [0.050238847794562086, 1.0069745427656598] 
 [0.0024964877994842166, 1.0039581459948161]
 [0.00012424350644497567, 1.000950784854063]
 [4.5854039847379484e-5, 0.999950334377818]
```

**Comment by ChrisRackauckas** [comment]
yeah if you can get it setup with destats. Also, that does look nice but normally we do `sol.destats` to show it. I am not sure we want to put more into the standard printout: it's already long.

**Comment by utkarsh530** [comment]
Yeah, I agree. I'll update the patch now and leave this for later. passing these values to `sol.destats` would be nice I am not sure how MATLAB passes this info it just prints it in the command line.

**Comment by ChrisRackauckas** [comment]
I think you can access it from MATLAB but I forget how.

**Pull request merged**